### PR TITLE
Reformat matrix_norm

### LINKED ISSUE #5172: matrix_norm

### PR DESCRIPTION
**Pull request by randomstate42**
fixes #5172

### CODE REVIEW

**Comment by randomstate42** [comment]
Close #5172

**Comment by randomstate42** [comment]
add_reformatting_checklist_linear_algebra

**Comment by randomstate42** [comment]
@juliagsy ready for review

**Comment by juliagsy** [comment]
Hey there! Thanks for contributing! Will review soonest possible!

**Comment by juliagsy** [comment]
Hey! As the PR is converted into draft, I'll review once its marked as ready for review! Thanks!

**Comment by randomstate42** [comment]
Hi @juliagsy , yes I converted it to draft because of a change in master branch (superset behaviour), but this change results in a bug so I opened an issue for that and will commit new examples once the new changes take place :construction: , Thanks.

**Comment by randomstate42** [comment]
@juliagsy can you please take a look at this PR #5420 so that I can proceed with this PR. Thanks

File: ivy/array/linear_algebra.py [hunk 1]
```diff
@@ -185,10 +185,41 @@ def matrix_norm(
         /,
         *,
         ord: Optional[Union[int, float, Literal[inf, -inf, "fro", "nuc"]]] = "fro",
-        axis: Optional[Union[int, Sequence[int]]] = None,
+        axis: Optional[Tuple[int, int]] = (-2, -1),
```
Comment (randomstate42) [inline ivy/array/linear_algebra.py:188]: I don't know why this type hint change still appears eventhough it was already updated in this [PR](#5420)

Comment (randomstate42) [reply ivy/array/linear_algebra.py:188]: #5420

Comment (juliagsy) [reply ivy/array/linear_algebra.py:188]: did you pull the latest commit in this PR?

**Comment by juliagsy** [comment]
Hi! I've commented in one of the comment thread above :)

**Comment by randomstate42** [comment]
I just found out that github UI shows the changes between the branches based on the best common ancestor and not the last version of the master branch, had to rebase the current branch(reformat_matrix_norm) to master. it 's ready for review now.

File: ivy/array/linear_algebra.py (hunk 1 above)
Comment (randomstate42) [reply ivy/array/linear_algebra.py]: done! now only shows the expected changes.

File: ivy/array/linear_algebra.py [hunk 2]
```diff
@@ -189,6 +189,37 @@ def matrix_norm(
         keepdims: bool = False,
         out: Optional[ivy.Array] = None,
     ) -> ivy.Array:
+        """
+        ivy.Array instance method variant of ivy.matrix_norm.
+        This method simply wraps the function, and so the docstring for
+        ivy.matrix_norm also applies to this method with minimal changes.
+        Parameters
+        ----------
+        x
```
Comment (juliagsy) [inline ivy/array/linear_algebra.py:198]: self

File: ivy/array/linear_algebra.py [hunk 3]
```diff
@@ -189,6 +189,37 @@ def matrix_norm(
...
+        axis
+            specifies the axes that hold 2-D matrices. Default: (-2, -1).
+        keepdims
+            If this is set to True, the axes which are normed over are left in
+            the result as dimensions with size one. With this option the result will
+            broadcast correctly against the original x. Default is False.
+        out
+            optional output array, for writing the result to. It must have a shape
+            that the inputs broadcast to.
+        Returns
```
Comment (juliagsy) [inline ivy/array/linear_algebra.py:212]: (for lint formatting tests) a blank line before the return subheading

File: ivy/array/linear_algebra.py [hunk 4]
```diff
@@ -189,6 +189,37 @@ def matrix_norm(
...
+        -------
+        ret
+            Matrix norm of the array at specified axes.
+        Examples
```
Comment (juliagsy) [inline ivy/array/linear_algebra.py:216]: (for lint formatting tests) a blank line before the examples subheading

File: ivy/array/linear_algebra.py [hunk 5]
```diff
@@ -189,6 +189,37 @@ def matrix_norm(
...
+        --------
+        >>> x = ivy.array([[1.1, 2.2, 3.3], [1.0, 2.0, 3.0]])
+        >>> y = x.matrix_norm(ord=1)
+        >>> print(y)
+        ivy.array(6.3)
```
Comment (juliagsy) [inline ivy/array/linear_algebra.py:221]: please also add examples for different values of the `axis` and `keepdims` args

File: ivy/container/linear_algebra.py [hunk 6]
```diff
@@ -912,6 +942,17 @@ def matrix_norm(
...
+        Examples
+        --------
+        >>> x = ivy.Container(a=ivy.array([[1.1, 2.2], [1., 2.]]), \
+                              b=ivy.array([[1., 2.], [3., 4.]]))
+        >>> y = x.matrix_norm(ord=1)
+        >>> print(y)
+        {
+            a: ivy.array(4.2),
+            b: ivy.array(6.)
+        }
```
Comment (juliagsy) [inline ivy/container/linear_algebra.py:954]: please also add examples to showcase different values of other args (`axis`, `keepdims`)

**Review by juliagsy — CHANGES_REQUESTED** [review]
Hey there! Great work so far! Just some small changes before officially merging the PR! As usual, please request my review when you're done! Thanks!

**Comment by juliagsy** [comment]
Thanks for contributing!

**Pull request merged**